lib/jazz_fingers/prompt.rb

```diff
@@ -32,44 +34,67 @@ def separator
       red_text(@separator)
     end
 
-    def name
-      blue_text(@application_name)
+    def context(module_name = "main")
+      name =
+        case module_name
+        when "main", "nil"
+          @application_name
+        when OBJECT_INSTANCE
+          abbreviated_context(module_name)
+        else
+          module_name
+        end
+
+      blue_text("(#{name})")
```

Comment:
Not sure i follow everything here but i guess once we figure out the other stuff i commented we can follow here.

Reply:
Good call-out thanks — I added some documentation in 4b815c7